### ISSUE TITLE
Docker image GCC 4.8 fails with 139 exit error

### ISSUE THREAD

**Issue by emiliogq** [docker-library/gcc#65]
# System details

* OS: Debian 10 (Buster)
* Docker version: Docker version 19.03.8, build afacb8b7f0
* Docker image: gcc:4.8 or gcc:4.8.5

# Steps to reproduce:

```
docker run --name gcc -it gcc:4.8.5
```
``` 
4.8.5: Pulling from library/gcc
47994b92ab73: Pull complete 
a3ed95caeb02: Pull complete 
9b7b75987c3c: Pull complete 
d66c4af59bfb: Pull complete 
26df7d6a7371: Pull complete 
d4aa50056c0d: Pull complete 
bfc6c0d1879f: Pull complete 
f7667401f7bb: Pull complete 
51ae28b91c03: Pull complete 
Digest: sha256:a515c9a2b8bf7d0f91470f7315be90862e249e6f4caa9f0d71898d3f2be8c280
Status: Downloaded newer image for gcc:4.8.5

```

I expected the bash shell but nothing else is prompted. Also, I've tried with:

```
docker run --name gcc -it gcc:4.8.5 /bin/bash
```

I inspected the container:

```
docker inspect gcc
```
I filtered the output:

```
"State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 139,
            "Error": "",
            "StartedAt": "2020-03-26T15:10:18.083255467Z",
            "FinishedAt": "2020-03-26T15:10:18.174373833Z"
        },

```

What I am doing wrong?

**Comment by emiliogq** [comment]
Also the docker events raise the following:

```
2020-03-26T17:01:54.251300827+01:00 container create 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (image=gcc:4.8.5, name=musing_tharp)
2020-03-26T17:01:54.251300827+01:00 container create 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (image=gcc:4.8.5, name=musing_tharp)
2020-03-26T17:01:54.252361283+01:00 container attach 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (image=gcc:4.8.5, name=musing_tharp)
2020-03-26T17:01:54.252361283+01:00 container attach 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (image=gcc:4.8.5, name=musing_tharp)
2020-03-26T17:01:54.271907156+01:00 network connect b95df2329dce2f15abc18e4b32eb36c8984378eda84260a85e3b4bda574010ef (container=7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7, name=bridge, type=bridge)
2020-03-26T17:01:54.271907156+01:00 network connect b95df2329dce2f15abc18e4b32eb36c8984378eda84260a85e3b4bda574010ef (container=7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7, name=bridge, type=bridge)
2020-03-26T17:01:54.667163606+01:00 2020-03-26T17:01:54.667163606+01:00 container start 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7container start 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (image=gcc:4.8.5, name=musing_tharp) (image=gcc:4.8.5, name=musing_tharp)

2020-03-26T17:01:54.668541228+01:00 container resize 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (height=52, image=gcc:4.8.5, name=musing_tharp, width=211)
2020-03-26T17:01:54.668541228+01:00 container resize 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (height=52, image=gcc:4.8.5, name=musing_tharp, width=211)
2020-03-26T17:01:54.768879831+01:00 2020-03-26T17:01:54.768879831+01:00 container die 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7container die 7ed6c0baac6081e9854f1dcde58a719d72db0849f1ad487a659da5afdf63fdf7 (exitCode=139, image=gcc:4.8.5, name=musing_tharp) (exitCode=139, image=gcc:4.8.5, name=musing_tharp)
```

Finally I test with another shell (sh) and the prompt showed up:

```
docker run --rm -it gcc:4.8.5 sh
#
```

**Comment by wglambert** [comment]
Similar to https://github.com/docker-library/mysql/issues/570
What do the `docker logs` show for the container?

I can't reproduce
```console
$ docker run --name gcc -it gcc:4.8.5
Unable to find image 'gcc:4.8.5' locally
4.8.5: Pulling from library/gcc
47994b92ab73: Pull complete 
a3ed95caeb02: Pull complete 
9b7b75987c3c: Pull complete 
d66c4af59bfb: Pull complete 
26df7d6a7371: Pull complete 
d4aa50056c0d: Pull complete 
bfc6c0d1879f: Pull complete 
f7667401f7bb: Pull complete 
51ae28b91c03: Pull complete 
Digest: sha256:a515c9a2b8bf7d0f91470f7315be90862e249e6f4caa9f0d71898d3f2be8c280
Status: Downloaded newer image for gcc:4.8.5

root@6202a118693b:/# gcc --version
gcc (GCC) 4.8.5
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Maybe trying it with `--security-opt apparmor=unconfined` to see if Apparmor is intervening

**Comment by emiliogq** [comment]
Do you have Debian 10 like me?

My workmate have another distro (Ubuntu) and it worked

docker logs do not output anything :(

**Comment by emiliogq** [comment]
I've just installed a Debian 10 VM, installed Docker inside an test the same and i got the following:

```
root@debian:~# docker run --rm -it gcc:4.8
Unable to find image 'gcc:4.8' locally
4.8: Pulling from library/gcc
Image docker.io/library/gcc:4.8 uses outdated schema1 manifest format. Please upgrade to a schema2 image for better future compatibility. More information at https://docs.docker.com/registry/spec/deprecated-schema-v1/
47994b92ab73: Pull complete 
a3ed95caeb02: Pull complete 
9b7b75987c3c: Pull complete 
d66c4af59bfb: Pull complete 
26df7d6a7371: Pull complete 
d4aa50056c0d: Pull complete 
bfc6c0d1879f: Pull complete 
f7667401f7bb: Pull complete 
51ae28b91c03: Pull complete 
Digest: sha256:43c6eaefe26ad414b2eefbee68688d1467823424cacbbd1703da4a50ce2c7654
Status: Downloaded newer image for gcc:4.8
```

**Comment by ksxauh** [comment]
The same for me. 
Just re-installed Linux, previously was Debian 9 all worked fine, now I've installed  Debian 10 and using the same `docker-compose.yml` I'm getting the same error with "Exited 130" and no any logs. Looks like this problem only on "Debian 10"

**Comment by emiliogq** [comment]
Yeah, I fixed using gcc:4.9 but I would like to use the previous one because some libraries were compiled with that version.

**Comment by yosifkit** [comment]
Since the old image is based on debian:wheezy, it is a classic case of: https://github.com/moby/moby/issues/28705

https://github.com/moby/moby/issues/28705#issuecomment-262268578
